xbmc/addons/AddonManager.cpp

```diff
@@ -479,7 +449,61 @@ bool CAddonMgr::HasAvailableUpdates()
   return !GetAvailableUpdates().empty();
 }
 
-bool CAddonMgr::GetAddons(const TYPE &type, VECADDONS &addons, bool enabled /* = true */)
+bool CAddonMgr::GetAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, true);
+}
+
+bool CAddonMgr::GetAddons(VECADDONS& addons, const TYPE& type)
+{
+  return GetAddonsInternal(type, addons, true);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, false);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons, const TYPE& type)
```

Comment:
Exactly. ADDON_UNKNOWN is used as a magic value to get 'any' type. Default values expose that information in the api. Not very obvious what `GetAddons(ADDON_UNKNOWN)` do, which is what you'll see when reading the calls.